### DISCUSSION TITLE
user-defined templates or shortcodes?

### DISCUSSION THREAD

**Discussion by hellt** [squidfunk/mkdocs-material#4467] [Q&A] [answered]
Hi all,
I somehow fill this is more related to mkdocs, but maybe you can point me in the right direction...

I am looking for a way to implement a user-based templating (which may be seen as a trivial shortcode application). Basically what I am trying to do is to have a separator between H2 elements that is constructed from HTML elements.

Here is how it looks like

```html
<div class="mdx-h2-divider">
<hr /><span class="twemoji vertical-middle"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><path d="M392.8 1.2c-17-4.9-34.7 5-39.6 22l-128 448c-4.9 17 5 34.7 22 39.6s34.7-5 39.6-22l128-448c4.9-17-5-34.7-22-39.6zm80.6 120.1c-12.5 12.5-12.5 32.8 0 45.3L562.7 256l-89.4 89.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l112-112c12.5-12.5 12.5-32.8 0-45.3l-112-112c-12.5-12.5-32.8-12.5-45.3 0zm-306.7 0c-12.5-12.5-32.8-12.5-45.3 0l-112 112c-12.5 12.5-12.5 32.8 0 45.3l112 112c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L77.3 256l89.4-89.4c12.5-12.5 12.5-32.8 0-45.3z"/></svg></span><hr />
</div>
```

What it does is that it render a separator between H2 headings:
<img width="856" alt="image" src="https://user-images.githubusercontent.com/5679861/194612778-35b6d41e-e68e-4aee-adef-75a43d24a59c.png">

Clearly, pasting this HTML blob between every H2 is cumbersome. Instead, I'd like to put in something like `{{my-separator}}` in the markdown and let mkdocs substitute it to the desired HTML blob.

Is this at all possible?

**Comment by squidfunk** [comment, marked as answer]
Maybe the [macros plugin](https://mkdocs-macros-plugin.readthedocs.io/en/latest/)?

**Reply by hellt** [reply]
Looks like it. Now I need to include the installation of this plugin to the publish pipeline for insiders
